### ISSUE TITLE
FileSystem Errors should subclass FsError

### ISSUE THREAD

**Issue by kurtkilpela** [GemTalk/FileSystemGs#123]
Right now, all FileSystem errors subclass Error. Instead, we should have them subclass a new class called FsError.

FsUnixError should remain in a separate hierarchy.

**Comment by kurtkilpela** [comment]
Fixed in 3ef2cc255b340a425d4dbad0dc9ad41d13dd5f23

**Comment by LisaAlmarode** [comment]
I think FsFileDescriptorInvalid should be under FsError too...

**Comment by kurtkilpela** [comment]
Yeah it should be. It also needs an error number associated with it.
